### ISSUE TITLE
Inconsistent training config

### ISSUE THREAD

**Issue by ArthurChen189** [MurtyShikhar/NNetnav#9]
Thank you for open-sourcing both the datasets and code!

I was trying to reproduce the training results. Looking at [finetune_with_accelerate](https://github.com/MurtyShikhar/NNetnav/blob/92ed927975cbca6982022dec1951a74437688640/finetuning/scripts/finetune_with_accelerate.sh), `max_seq_length=4096`, `epochs=5`, where in the paper the context window is 20_000. `epochs=2`. Are the rest of the arguments consistent with the ones in the paper for SFT?

**Comment by MurtyShikhar** [comment]
Yes thats a very astute observation. 2 comments:
1. Use the latest version of open-instruct. The one here is a super old version (that i pushed longtime ago)

2. Heres a config you can use (pls modify according to your setup but note the hyperparams):


export CUDA_VISIBLE_DEVICES=0,1,2,3


MODEL_SIZE=8B
NUM_GPUS=4
BATCH_SIZE_PER_GPU=1
TOTAL_BATCH_SIZE=32
GRADIENT_ACC_STEPS=$(($TOTAL_BATCH_SIZE/$NUM_GPUS/$BATCH_SIZE_PER_GPU))
MODEL_PATH="/scr/biggest/smurty/finetuned_webagents/finetuning/base/models--meta-llama--Meta-Llama-3.1-8B-Instruct/"
echo "Training llama model ${MODEL_SIZE} using $NUM_GPUS GPUs, $BATCH_SIZE_PER_GPU batch size per GPU, $GRADIENT_ACC_STEPS gradient accumulation steps"
# You can also set --gradient_checkpointing or use `stage3_offloading_accelerate.conf` to save memory,
# but it will trade off speed.
accelerate launch \
    --mixed_precision bf16 \
    --num_machines 1 \
    --num_processes $NUM_GPUS \
    --use_deepspeed \
    --deepspeed_config_file configs/ds_configs/stage3_offloading_accelerate.conf \
    open_instruct/finetune.py \
    --model_name_or_path $MODEL_PATH \
    --use_flash_attn \
    --gradient_checkpointing \
    --tokenizer_name $MODEL_PATH \
    --use_slow_tokenizer \
    --train_file <path-to-train-jsonl> \
    --max_seq_length 20000 \
    --preprocessing_num_workers 128 \
    --per_device_train_batch_size $BATCH_SIZE_PER_GPU \
    --gradient_accumulation_steps $GRADIENT_ACC_STEPS \
    --learning_rate 2e-5 \
    --lr_scheduler_type linear \
    --warmup_ratio 0.03 \
    --weight_decay 0. \
    --num_train_epochs 2 \
    --output_dir <your-path>\
    --with_tracking \
    --logging_steps 1 \
    --checkpointing_steps 500 \
    --report_to wandb \
    --wandb_entity <your-name> \
    --exp_name Llama-3-1-8B-Instruct_nnetnav_wa-and-ow\

**Comment by ArthurChen189** [comment]
Great! Thanks!
